pkg/sql/exec_log.go

```diff
@@ -165,7 +165,8 @@ func (p *planner) maybeLogStatementInternal(
 	maxEventFrequency := TelemetryMaxEventFrequency.Get(&p.execCfg.Settings.SV)
 
 	// We only consider non-internal SQL statements for telemetry logging.
-	telemetryLoggingEnabled := telemetryLoggingEnabled.Get(&p.execCfg.Settings.SV) && execType != executorTypeInternal
+	telemetryLoggingEnabled := telemetryLoggingEnabled.Get(&p.execCfg.Settings.SV) &&
```

Comment:
nit: Consider updating the above comment, i.e. `..., unless telemetryInternalQueriesEnabled is true.`